src/theory/quantifiers/sygus/sygus_unif_rl.cpp

```diff
@@ -804,6 +804,64 @@ Node SygusUnifRl::DecisionTreeInfo::buildSolMinCond(Node cons,
       // this violates the invariant that the i^th conditional enumerator
       // resolves the i^th separation conflict
       exp_conflict = true;
+      // Since the explanation of the condition (c_exp above) does not account
+      // for builtin subterms, we additionally require that the valuation of
+      // the condition is indeed different on the two points.
+      // For example, say ce has model value equal to the SyGuS datatype term:
+      //   C_leq_xy( 0, 1 )
+      // where C_leq_xy is a SyGuS datatype constructor taking two integer
+      // constants c_x and c_y, and whose builtin version is:
+      //   (0*x + 1*y >= 0)
+      // Then, c_exp above is:
+      //   is-C_leq_xy( ce )
+      // which is added to our explanation of the conflict, which does not
+      // account for the values of the arguments of C_leq_xy.
+      // Now, say that we are in a separation conflict due to f(1,2) and f(2,3)
+      // being assigned different values; the value of ce does not separate
+      // these two terms since:
+      //   (y>=0) { x -> 1, y -> 2 } = (y>=0) { x -> 2, y -> 3 } = true
+      // The code below adds a constraint that states that the above values are
+      // the same, which is part of the reason for the conflict. In the above
+      // example, we generate:
+      //   (DT_SYGUS_EVAL ce 1 2) == (DT_SYGUS_EVAL ce 2 3) { ce -> M(ce) }
+      // which unfolds via the SygusEvalUnfold utility to:
+      //   ( (c_x ce)*1 + (c_y ce)*2 >= 0 ) == ( (c_x ce)*2 + (c_y ce)*3 >= 0 )
+      // where c_x and c_y are the selectors of the subfields of C_leq_xy.
+      SygusTypeInfo& ti = d_unif->d_tds->getTypeInfo(ce.getType());
+      // The above is only necessary if we use symbolic constructors.
+      if (ti.hasSubtermSymbolicCons())
```

Comment:
Could you move this test to just below `ex_conflict = true` and break in case it's not a symbolic cons?